Update `contrast-color()` for removal of `max` parameter

weinig opened the pull request:
#### 818477725b452ce1f2b5bab79e9a70ae49fe5ff0
<pre>
Update `contrast-color()` for removal of `max` parameter
<a href="https://bugs.webkit.org/show_bug.cgi?id=287410">https://bugs.webkit.org/show_bug.cgi?id=287410</a>

Reviewed by Tim Nguyen.

Updates the implementation of `contrast-color()` to match the latest
spec which removed `max` parameter (see <a href="https://github.com/w3c/csswg-drafts/pull/11654).">https://github.com/w3c/csswg-drafts/pull/11654).</a>

Also updates tests and moves the non-WPT tests to WPT, changing the
computed style tests to allow either `white` or `black` for all computed
values due to the lack of agreed upon algorithm.

To keep coverage of the algorithm in WebKit, a copy of the computed style
test was kept in css/fast that retains the specific results.

* LayoutTests/fast/css/color-computed-contrast-color-function-expected.txt:
* LayoutTests/fast/css/color-computed-contrast-color-function.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-contrast-color-function-expected.txt: Copied from LayoutTests/fast/css/color-computed-contrast-color-function-expected.txt.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-contrast-color-function.html: Copied from LayoutTests/fast/css/color-computed-contrast-color-function.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-invalid-contrast-color-function-expected.txt: Renamed from LayoutTests/fast/css/color-invalid-contrast-color-function-expected.txt.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-invalid-contrast-color-function.html: Renamed from LayoutTests/fast/css/color-invalid-contrast-color-function.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-contrast-color-function-expected.txt: Renamed LayoutTests/fast/css/color-valid-contrast-color-function-expected.txt.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-contrast-color-function.html: Renamed LayoutTests/fast/css/color-valid-contrast-color-function.html.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.cpp:
* Source/WebCore/css/values/color/CSSContrastColor.cpp:
* Source/WebCore/css/values/color/CSSContrastColor.h:
* Source/WebCore/css/values/color/CSSContrastColorResolver.h:
* Source/WebCore/css/values/color/CSSContrastColorSerialization.h:
* Source/WebCore/style/values/color/StyleContrastColor.cpp:
* Source/WebCore/style/values/color/StyleContrastColor.h:

Canonical link: <a href="https://commits.webkit.org/290210@main">https://commits.webkit.org/290210@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5d1307b42020a47416edbefc3247de1e450a056b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/89159 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8683 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/43986 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/94140 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/39920 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/91210 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/9071 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/17052 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/68703 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/26373 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/92161 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/6951 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/80922 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/49065 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/6700 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/35300 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/39027 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/77063 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/36286 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/95973 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/16339 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/12099 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/77580 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/16595 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/76708 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/76871 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18984 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/21276 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/19886 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/9453 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/16353 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/21664 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/16094 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/19545 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/17875 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-contrast-color-function.html:
```diff
@@ -0,0 +1,51 @@
+</div>
+<script>
+
+// NOTE: As the algorithm for determining constrast is not defined, all computed values can validly be either `rgb(0, 0, 0)` or `rgb(255, 255, 255)`.
+
+// Test basic legacy colors
+test_computed_value("background-color", "contrast-color(white)", ["rgb(0, 0, 0)", "rgb(255, 255, 255)"]);
+test_computed_value("background-color", "contrast-color(black)", ["rgb(0, 0, 0)", "rgb(255, 255, 255)"]);
```
There's probably some obvious ones like `contrast-color(white)` / `contrast-color(black)` where we can assume are `black` / `white`. Given this requirement:

> However, colors returned by this function should still meet the WCAG 2.1 [section 1.4.3 Contrast (Minimum)](https://www.w3.org/TR/WCAG21/#contrast-minimum) for AA large text, as many authors need to meet legal requirements that mandate this.

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-contrast-color-function.html:
```diff
@@ -0,0 +1,51 @@
+// Test nested contrast-color
+test_computed_value("background-color", "contrast-color(contrast-color(pink))", ["rgb(0, 0, 0)", "rgb(255, 255, 255)"]);
+
+// Test currentColor
+test_computed_value("background-color", "contrast-color(currentcolor)", ["rgb(0, 0, 0)", "rgb(255, 255, 255)"]);
```
might be more useful to test that it's equal to contrast-color(pink)

weinig committed:
Update `contrast-color()` for removal of `max` parameter
https://bugs.webkit.org/show_bug.cgi?id=287410

Reviewed by Tim Nguyen.

Updates the implementation of `contrast-color()` to match the latest
spec which removed `max` parameter (see https://github.com/w3c/csswg-drafts/pull/11654).

Also updates tests and moves the non-WPT tests to WPT, changing the
computed style tests to allow either `white` or `black` for all computed
values due to the lack of agreed upon algorithm.

To keep coverage of the algorithm in WebKit, a copy of the computed style
test was kept in css/fast that retains the specific results.

* LayoutTests/fast/css/color-computed-contrast-color-function-expected.txt:
* LayoutTests/fast/css/color-computed-contrast-color-function.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-contrast-color-function-expected.txt: Copied from LayoutTests/fast/css/color-computed-contrast-color-function-expected.txt.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-contrast-color-function.html: Copied from LayoutTests/fast/css/color-computed-contrast-color-function.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-invalid-contrast-color-function-expected.txt: Renamed from LayoutTests/fast/css/color-invalid-contrast-color-function-expected.txt.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-invalid-contrast-color-function.html: Renamed from LayoutTests/fast/css/color-invalid-contrast-color-function.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-contrast-color-function-expected.txt: Renamed LayoutTests/fast/css/color-valid-contrast-color-function-expected.txt.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-contrast-color-function.html: Renamed LayoutTests/fast/css/color-valid-contrast-color-function.html.
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.cpp:
* Source/WebCore/css/values/color/CSSContrastColor.cpp:
* Source/WebCore/css/values/color/CSSContrastColor.h:
* Source/WebCore/css/values/color/CSSContrastColorResolver.h:
* Source/WebCore/css/values/color/CSSContrastColorSerialization.h:
* Source/WebCore/style/values/color/StyleContrastColor.cpp:
* Source/WebCore/style/values/color/StyleContrastColor.h:

Canonical link: https://commits.webkit.org/290210@main
